sdc/datatypes/common_functions.py

```diff
@@ -189,6 +189,62 @@ def _append_list_string_array_impl(A, B):
             return _append_list_string_array_impl
 
 
+@register_jitable
+def fill_array(data, size, fill_value=numpy.nan, push_back=True):
+    """
+    Fill array with given values to reach the size
+    """
+
+    if push_back:
+        return numpy.append(data, numpy.repeat(fill_value, size - data.size))
+
+    return numpy.append(numpy.repeat(fill_value, size - data.size), data)
+
+
+@register_jitable
+def fill_str_array(data, size, push_back=True):
+    """
+    Fill StringArrayType array with given values to reach the size
+    """
+
+    string_array_size = len(data)
+    none_array_size = size - string_array_size
+    num_chars = sdc.str_arr_ext.num_total_chars(data)
+
+    result_data = sdc.str_arr_ext.pre_alloc_string_array(size, num_chars)
+
+    # Keep NaN values of initial array
+    arr_is_na_mask = numpy.array([sdc.hiframes.api.isna(data, i) for i in
+                                  numba.prange(string_array_size)])
+    data_str_list = sdc.str_arr_ext.to_string_list(data)
+    nan_list = [''] * none_array_size
+
+    result_list = data_str_list + nan_list if push_back else nan_list + data_str_list
+    sdc.str_arr_ext.cp_str_list_to_array(result_data, result_list)
+
+    # Batch=64 iteration to avoid threads competition
+    batch_size = 64
+    if push_back:
+        for i in numba.prange(string_array_size//batch_size + 1):
+            for j in range(i, max(i + batch_size, string_array_size)):
+                if arr_is_na_mask[j]:
+                    str_arr_set_na(result_data, j)
+        for i in numba.prange(none_array_size//batch_size + 1):
```

Comment:
This would work only in case `string_array_size` is multiple to 64. But if it is not?